docs/source/reference/respawn/weapon.rst

```diff
@@ -5,53 +5,327 @@ Methods of the weapon class
+
+    .. cpp:function:: void RegenerateAmmoReset()
+
+    .. cpp:function:: float GetCoreDuration()
+
+        Titan core duration
+
+    Retrieve Weapon Keyvalues
+    -------------------------
+
+    .. cpp:function:: var GetWeaponInfoFileKeyField( string field )
+
+        Returns the content of the passed keyvalue field. Use `expect` to specify the return type at compile time.
+
+    .. cpp:function:: int GetWeaponDamageFlags()
+
+    .. cpp:function:: bool GetAllowHeadShots()
+
+        Wether or not the weapon is able to score critical headshot damage.
+    
+    .. cpp:function:: int GetAmmoPerShot()
+
+    .. cpp:function:: int GetProjectilesPerShot()
+
+        The Amount of individual projectiles that are fired by the weapon.
+
+    .. cpp:function:: float GetMaxDamageFarDist()
+
+        The maximal distance in which the weapon is able to deal damage.
+
+    .. cpp:function:: bool ShouldPredictProjectiles()
+
+    Weapon Settings
+    ---------------
+
+    .. cpp:function:: bool GetWeaponSettingBool( unknown )
+
+    .. cpp:function:: int GetWeaponSettingInt( unknown )
+
+    .. cpp:function:: int GetWeaponSettingFloat( unknown )
+
+    .. cpp:function:: int GetWeaponSettingString( unknown )
+
+    Projectile Settings
+    -------------------
+
+    .. cpp:function:: float GetProjectileWeaponSettingFloat( unknown)
+
+    .. cpp:function:: int GetProjectileWeaponSettingInt( unknown )
+
+    .. cpp:function:: float GetProjectileWeaponSettingFloat( unknown )
+
+    .. cpp:function:: int GetProjectileWeaponSettingString( unknown )
+
+    Clip Size
+    ---------
+
     .. cpp:function:: int GetWeaponPrimaryClipCount()
-        
+
     .. cpp:function:: int GetWeaponPrimaryClipCountMax()
-        
-    .. cpp:function:: bool GetWeaponSettingBool()
-        
-    .. cpp:function:: float GetWeaponSettingFloat()
-        
-    .. cpp:function:: int GetWeaponSettingInt()
-        
-    .. cpp:function:: unknown GetWeaponType()
-        
-    .. cpp:function:: unknown GetXP()
-        
-    .. cpp:function:: unknown SetNextAttackAllowedTime()
-        
+
+    .. cpp:function:: void SetWeaponPrimaryClipCount( int clipsize )
+
+    .. cpp:function:: void SetWeaponPrimaryClipCountAbsolute( int clipsize )
+
+    .. cpp:function:: void SetWeaponPrimaryClipCountNoRegenReset( int clipsize )
+
+    Reloading
+    ---------
+
+    .. cpp:function:: bool IsReloading()
+
+    .. cpp:function:: int GetReloadMilestoneIndex()
+
+        Reload Progress. Only used on the LStar
+
+    ADS
+    ---
+
+    .. cpp:function:: bool IsWeaponInAds()
+
+        Returns `true` if the gun is aimed down sights.
+    
+    .. cpp:function:: bool IsWeaponAdsButtonPressed()
+
+        Returns `true` as long as the button set up for aiming is pressed.
+
+    Forced ADS
+    ----------
+
+    .. cpp:function:: void SetForcedADS()
+
+    .. cpp:function:: bool GetForcedADS()
+
+    .. cpp:function:: void ClearForcedADS()
+
+    Weapon Mods
+    -----------
```

Comment:
I'm talking about these subsections I added to make categorizing the functions into different purposes easier for me. I thought they might be useful to see at a glance what's possible with the weapon entity

Reply:
Ah no those are working as intended. I'll just merge this, and if you want to change it at a later time you can always just make another PR. ^^